src/fideslang/models.py

```diff
@@ -285,23 +298,150 @@ class MyDatasetField(DatasetFieldBase):
     )
 
 
-class DatasetField(DatasetFieldBase):
+EdgeDirection = Literal["from", "to"]
+
+
+class FidesDatasetReference(BaseModel):
+    """Reference to a field from another Collection"""
+
+    dataset: FidesKey
+    field: str
+    direction: Optional[EdgeDirection]
+
+
+class FidesMeta(BaseModel):
+    """Annotations used for query traversal"""
+
+    references: Optional[List[FidesDatasetReference]] = None
+    identity: Optional[str]
+    primary_key: Optional[bool]
+    data_type: Optional[str]
+    """Optionally specify the data type. Fides will attempt to cast values to this type when querying."""
+    length: Optional[PositiveInt]
+    """Optionally specify the allowable field length. Fides will not generate values that exceed this size."""
+    return_all_elements: Optional[bool]
+    """Optionally specify to query for the entire array if the array is an entrypoint into the node. Default is False."""
+    read_only: Optional[bool]
```

Comment:
```suggestion
    read_only: Optional[bool] = Field(description="Optionally specify to query for the entire array if the array is an entrypoint into the node. Default is False.")
```

I _think_ this is the right syntax 

I believe these should all be included as descriptions instead of code comments so we can use them later for auto-generated docs as well as other potential things